CUJExecutor/manipulateProfilesAndTest

```diff
@@ -14,60 +14,123 @@
 #See the License for the specific language governing permissions and
 #limitations under the License.
 
-INPUT_FILE=${1:-input.txt}
-INCLUDE_MEASURED_CUJ=${2?"ERROR: need to specify whether the measured CUJ should be included in the profile"}
-FILE_NAME="${INPUT_FILE%.*}"
-PRE_CUJ=$(sed '1q;d' $INPUT_FILE)
-POST_CUJ=$(sed '2q;d' $INPUT_FILE)
-RECORD=$(sed '4q;d' $INPUT_FILE)
-
 
+##compiles the app according to the COMPILATION_FLAG and then measures the cuj specified by the INPUT_FILE 
+##stores a command/output summary, a short data summary, the profile used to compile the app (if applicable) 
+##and a recording of the median run 
 
-##FORMAT PRE AND POS CUJS
-PRE_CUJ=${PRE_CUJ//\‘/\'}
-PRE_CUJ=${PRE_CUJ//\’/\'}
-POST_CUJ=${POST_CUJ//\‘/\'}
-POST_CUJ=${POST_CUJ//\’/\'}
 
-#use escape char to allow special characters (which would terminate the argument in the am instrument command or do other undesirable things).
-SpecialChars=(";"  "'" "&" "%" "(" ")" "^" "#" "@" "!" "+")
-for char in ${SpecialChars[@]}; do
-	exp="\\${char}"
-	PRE_CUJ=${PRE_CUJ//$exp/\\$char}
-	POST_CUJ=${POST_CUJ//$exp/\\$char}
-done
-# Wrap strings in \" to allow for spaces within them. 
-PRE_CUJ="\\\"$PRE_CUJ\\\""
-POST_CUJ="\\\"$POST_CUJ\\\""
 
+#PARSING INPUTS:
+INPUT_FILE=${1:-input.txt}
+COMPILATION_FLAG=${2?"ERROR: need to specify a compilation flag"} 
+if [ $COMPILATION_FLAG != "e" ] && [ $COMPILATION_FLAG != "p" ] && [ $COMPILATION_FLAG != "c" ] && [ $COMPILATION_FLAG != "a" ] 
+then
+	echo "compilation flag invalid. Use one of {e - empty profile, p - compile actions in Pre-CUJ, c - compile entire CUJ, a - compile all code}"
+	exit 1	
+fi	
+
+#Initializing variables
+FILE_NAME="${INPUT_FILE##*/}"
+FILE_NAME="${FILE_NAME%.*}"
+RECORD=$(sed '4q;d' $INPUT_FILE)
+TIME_DATA="/tmp/time_data.txt"
+
+#PREPARING OUTPUTS:
```

Comment:
I highly recommend switching to python, java, or go for scripting.  This script is becoming untestable.